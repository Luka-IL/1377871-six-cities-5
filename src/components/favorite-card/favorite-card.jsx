import React from "react";
import PropTypes from "prop-types";
import withFavoriteButton from "../../hocs/with-favorite-button/with-favorite-button";
import FavoriteButtonComponent from "../favorite-button-offer-card/favorite-button-offer-card";
import {ActionCreator} from "../../store/action";
import {connect} from "react-redux";
import {AppRoute} from "../../const";
import {PropTypesOffer} from "../../proptypes";

const FavoriteButton = withFavoriteButton(FavoriteButtonComponent);

const FavoriteCard = (props) => {

  const {offer, onOfferClick, changeCity, activateOffer, redirectMain} = props;
  const {id, city, price, image, rating, type, title, favorite} = offer;

  const onOfferCardClick = (evt) => {
    evt.preventDefault();
    changeCity(city.name);
    activateOffer(offer);
    onOfferClick(`${AppRoute.OFFER}/${id}`);
  };

  const redirectToMainCity = (evt) => {
    evt.preventDefault();
    changeCity(city.name);
    redirectMain(AppRoute.ROOT);
  };

  return (
    <li className="favorites__locations-items" key={id}>
      <div className="favorites__locations locations locations--current">
        <div className="locations__item">
          <a className="locations__item-link" onClick={redirectToMainCity}>
            <span>{city.name}</span>
          </a>
        </div>
      </div>
      <div className="favorites__places">
        <article className="favorites__card place-card">
          <div className="favorites__image-wrapper place-card__image-wrapper" onClick={onOfferCardClick}>
            <a href="#">
              <img className="place-card__image" src={image} width="150" height="110" alt="Place image" />
            </a>
          </div>
          <div className="favorites__card-info place-card__info">
            <div className="place-card__price-wrapper">
              <div className="place-card__price">
                <b className="place-card__price-value">&euro;{price}</b>
                <span className="place-card__price-text">&#47;&nbsp;night</span>
              </div>
              <FavoriteButton
                favorite={favorite}
                id={id}
              />
            </div>
            <div className="place-card__rating rating">
              <div className="place-card__stars rating__stars">
                <span style={{width: (rating * 20) + `%`}}></span>
                <span className="visually-hidden">Rating</span>
              </div>
            </div>
            <h2 className="place-card__name" onClick={onOfferCardClick}>
              <a href="#">{title}</a>
            </h2>
            <p className="place-card__type">{type}</p>
          </div>
        </article>
      </div>
    </li>
  );
};

FavoriteCard.propTypes = {
  onOfferClick: PropTypes.func.isRequired,
  changeCity: PropTypes.func.isRequired,
  redirectMain: PropTypes.func.isRequired,
  activateOffer: PropTypes.func.isRequired,
  offer: PropTypesOffer
};

const mapDispatchToProps = (dispatch) => ({
  redirectMain(url) {
    dispatch(ActionCreator.redirectToRoute(url));
  },
  changeCity(city) {
    dispatch(ActionCreator.changeCity(city));
  }
});

export {FavoriteCard};
export default connect(null, mapDispatchToProps)(FavoriteCard);
