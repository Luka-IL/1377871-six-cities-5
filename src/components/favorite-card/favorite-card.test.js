import React from "react";
import renderer from "react-test-renderer";
import {FavoriteCard} from "./favorite-card";
import configureStore from "redux-mock-store";
import {Provider} from "react-redux";

const mockStore = configureStore([]);

const store = mockStore({
  DATA: {
    active:
      {
        title: `1Beautiful & luxurious studio at great location`,
        premum: false,
        type: `apartment`,
        price: 20,
        rating: 4
      },
  },
  USER: {
    authorizationStatus: `AUTH`
  }
});

const offer = {
  city: {
    name: `Amsterdam`
  },
  title: `1Beautiful & luxurious studio at great location`,
  premium: false,
  type: `apartment`,
  price: 20,
  rating: 4,
  image: ``
};

it(`Should FavoriteCard render correctly`, () => {
  const tree = renderer
    .create(
        <Provider store={store}>
          <FavoriteCard
            onOfferClick ={() => {}}
            offer={offer}
          />
        </Provider>
    ).toJSON();

  expect(tree).toMatchSnapshot();
});
