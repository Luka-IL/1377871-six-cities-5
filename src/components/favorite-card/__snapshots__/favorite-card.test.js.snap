// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should FavoriteCard render correctly 1`] = `
<li
  className="favorites__locations-items"
>
  <div
    className="favorites__locations locations locations--current"
  >
    <div
      className="locations__item"
    >
      <a
        className="locations__item-link"
        onClick={[Function]}
      >
        <span>
          Amsterdam
        </span>
      </a>
    </div>
  </div>
  <div
    className="favorites__places"
  >
    <article
      className="favorites__card place-card"
    >
      <div
        className="favorites__image-wrapper place-card__image-wrapper"
        onClick={[Function]}
      >
        <a
          href="#"
        >
          <img
            alt="Place image"
            className="place-card__image"
            height="110"
            src=""
            width="150"
          />
        </a>
      </div>
      <div
        className="favorites__card-info place-card__info"
      >
        <div
          className="place-card__price-wrapper"
        >
          <div
            className="place-card__price"
          >
            <b
              className="place-card__price-value"
            >
              €
              20
            </b>
            <span
              className="place-card__price-text"
            >
              / night
            </span>
          </div>
          <button
            className="place-card__bookmark-button button"
            type="button"
          >
            <svg
              className="place-card__bookmark-icon"
              height="19"
              onClick={[Function]}
              width="18"
            >
              <use
                xlinkHref="#icon-bookmark"
              />
            </svg>
            <span
              className="visually-hidden"
            >
              To bookmarks
            </span>
          </button>
        </div>
        <div
          className="place-card__rating rating"
        >
          <div
            className="place-card__stars rating__stars"
          >
            <span
              style={
                Object {
                  "width": "80%",
                }
              }
            />
            <span
              className="visually-hidden"
            >
              Rating
            </span>
          </div>
        </div>
        <h2
          className="place-card__name"
          onClick={[Function]}
        >
          <a
            href="#"
          >
            1Beautiful & luxurious studio at great location
          </a>
        </h2>
        <p
          className="place-card__type"
        >
          apartment
        </p>
      </div>
    </article>
  </div>
</li>
`;
